tests/test_stunnel.py

```diff
@@ -0,0 +1,188 @@
+    **_stunnel_kwargs,
+    extra_environment_variables={
+        "STUNNEL_DEBUG": "info",
+        "STUNNEL_SERVICE_NAME": "python",
+        "STUNNEL_ACCEPT": "0.0.0.0:8443",
+        "STUNNEL_CONNECT": "0.0.0.0:8000",
+    },
+)
+
+_PYTHON_WEB_SERVER = DerivedContainer(
+    base="registry.suse.com/bci/python",
+    containerfile="""
+RUN echo "hello world" > index.html
+ENTRYPOINT ["python3", "-m", "http.server"]
+""",
+)
+
+
+_STUNNEL_POD = pytest.param(
+    Pod(
+        containers=[_PYTHON_WEB_SERVER, _STUNNEL_PYTHON_HTTP_TUNNEL_CTR],
+        forwarded_ports=[PortForwarding(container_port=8443)],
+    ),
+    marks=_marks,
+)
+
+
+@pytest.mark.parametrize("pod", [_STUNNEL_POD], indirect=True)
+def test_stunnel_http_proxy(pod: PodData):
+    """Smoke test for stunnel to wrap a python http server in a TLS connection
+    using the generated certificate. The python web server is running in its own
+    container and stunnel in another one both in a single pod that only exposes
+    the TLS endpoint.
+
+    """
+
+    @tenacity.retry(
+        stop=tenacity.stop_after_attempt(5), wait=tenacity.wait_exponential()
+    )
+    def get_request_to_stunnel(route: str = "") -> requests.Response:
+        resp = requests.get(
+            f"https://127.0.0.1:{pod.forwarded_ports[0].host_port}/{route}",
+            verify=_SERVER_PEM,
+            timeout=5,
+        )
+        resp.raise_for_status()
+        return resp
+
+    assert "hello world" in get_request_to_stunnel().text
+
+
+_CTR_PORT = 8000
+
+
+def _create_http_forward_ctr(log_level: str = "info") -> ParameterSet:
+    return pytest.param(
+        DerivedContainer(
+            **_stunnel_kwargs,
+            forwarded_ports=[PortForwarding(container_port=_CTR_PORT)],
+            extra_environment_variables={
+                "STUNNEL_DEBUG": log_level,
+                "STUNNEL_SERVICE_NAME": "neverssl",
+                "STUNNEL_ACCEPT": f"0.0.0.0:{_CTR_PORT}",
+                "STUNNEL_CONNECT": "neverssl.com:80",
+            },
+        ),
+        marks=_marks,
+        id=STUNNEL_CONTAINER.id,
+    )
+
+
+@pytest.mark.parametrize(
+    "container", [_create_http_forward_ctr()], indirect=True
+)
+def test_http_tunnel_to_neverssl_com(container: ContainerData) -> None:
+    """Simple smoke test of stunnel wrapping the HTTP connection to
+    neverssl.com with TLS.
+
+    """
+
+    @tenacity.retry(
+        stop=tenacity.stop_after_attempt(5), wait=tenacity.wait_exponential()
+    )
+    def get_request_to_stunnel(route: str = "") -> requests.Response:
+        resp = requests.get(
+            f"https://127.0.0.1:{container.forwarded_ports[0].host_port}/{route}",
+            verify=_SERVER_PEM,
+            timeout=5,
+        )
+        resp.raise_for_status()
+        return resp
+
+    resp = get_request_to_stunnel()
+    assert resp.status_code == 200
+    assert (
+        "This website is for when you try to open Facebook, Google, Amazon, etc"
```

Comment:
```suggestion
        assert "neverssl" in resp.text.lower()
```
Making assumptions on external resources we have no control over is always difficult. I would limit the assumption to a minimum, so that the test doesn't fail when someone slightly changes the text on neverssl.com. We already check for status code 200, I think this check doesn't need to be extensive here.